matrix/matrix.py

```diff
@@ -0,0 +1,198 @@
+import operator
+from itertools import repeat
+
+
+class Matrix:
+    """An immutable object representing a matrix data-type. This
+    implementation only supports even-side matrices (n * n).
+
+    Attributes:
+        _matrix (tuple): contains the data about the matrix. Each line of
+                         the matrix is represented by a tuple.
+    """
+    __slots__ = '_matrix'
+
+    def __init__(self, *lines):
+        if not self._is_valid_data(*lines):
+            raise ValueError('Data is not valid')
+
+        super(Matrix, self).__setattr__('_matrix', tuple(lines))
+
+    def __setattr__(self, key, value):
+        """Defines that Matrix attributes are immutable."""
+        raise AttributeError(f'{self.__class__} instance cannot be modified')
+
+    @staticmethod
+    def _is_valid_data(*data: tuple) -> bool:
+        """Checks if the data given is valid. Valid data is defined if it
+        contains tuples of the same length (Matrix only supports n * n) and
+        their value is an integer."""
+        # check if each line is a tuple.
+        if not all([isinstance(line_data, tuple) for line_data in data]):
+            return False
+
+        # check if each item in each line is an int.
+        if not all([any([isinstance(item, int), isinstance(item, float)]) for
+                    line_data in data for item in line_data]):
+            return False
+
+        # checks if the given data is of a n * n matrix.
+        if not all([len(line) == len(data) for line in data]):
+            return False
+
+        return True
+
+    @classmethod
+    def unity(cls, number_lines: int):
+        """Generates a matrix with a main diagonal of 1's, everything else
+        is 0's.
+        """
+        tuples_in_chars = [
+            list(str('1' + ('0' * i)).zfill(number_lines)[::-1]) for i in
+            range(number_lines)]
+
+        return cls(*tuple(
+            [tuple(map(int, tuple_chars)) for tuple_chars in tuples_in_chars]))
+
+    @classmethod
+    def ones(cls, number_lines):
+        """Generates a matrix full of 1's."""
+        return cls(*tuple(
+            [tuple(repeat(1, number_lines)) for _ in range(number_lines)]))
```

Comment:
You can repeat the lines just like you used `repeat` to repeat the `1`
Alternatively, `(1,) * number_lines` works just the same